### PR TITLE
feat: integrates mongoose

### LINKED ISSUE #100: Integrate mongoose
## Expected Behavior
Mongoose should be integrated.
Connection to database should be done via environment files.

## Actual Behavior
No ODM

## Steps to Reproduce the Problem
N/A

## Specifications
N/A

### PR DESCRIPTION
**Pull request by guilhermevrs**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: Closes #100 

## What is the new behavior?

- mongoose connection
- environment variables

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: packages/server/src/app.module.ts [hunk 1]
```diff
@@ -1,9 +1,34 @@
...
+      imports: [ConfigModule],
+      useFactory: async (configService: ConfigService) => {
+        let uri = configService.get<string>('DB_URL');
+        if (!uri) {
+          uri = `mongodb://${configService.get<string>(
+            'MONGO_ADMIN_USR',
+          )}:${configService.get<string>(
+            'MONGO_ADMIN_PWD',
+          )}@localhost:${configService.get<string>('MONGO_PORT')}`;
+        }
```
Comment (tib-tib) [inline packages/server/src/app.module.ts:24]: I think this if is not needed. In local in your .env file you could have

```
DB_URL="mongodb://${MONGO_ADMIN_USR}:${MONGO_ADMIN_PWD}@localhost:${MONGO_PORT}"
```
so I don't think this logic is needed

Comment (guilhermevrs) [reply packages/server/src/app.module.ts:24]: Before doing it, Ive tried on several ways of reusing the env variables in it, but always got it the literal strings

image

Comment (guilhermevrs) [reply packages/server/src/app.module.ts:24]: It's apparently a limitation from dotenv package (used behind the scenes).
We could, though, use https://github.com/motdotla/dotenv-expand to expand those.

Let's go with this approach?
( I am not a fan of rebuilding the string on the server either)

Comment (guilhermevrs) [reply packages/server/src/app.module.ts:24]: AH!
Nevermind, there is an option for it on the ConfigModule from nestjs.

I am updating it right away

**Review by tib-tib — APPROVED** [review]

**Review by tib-tib — APPROVED** [review]

**Pull request merged**